controllers/storagecluster/reconcile.go

```diff
@@ -394,6 +396,19 @@ func (r *StorageClusterReconciler) reconcilePhases(
 		}
 	}
 
+	// if the StorageCluster is configured to run on non-default host network, rook-ceph-operator needs to run on host network as well
+	// this is validated by checking if the AddressRanges.Public field is set in the Network spec
+	// since pod network cannot always communicate with non-default host network rook-ceph-operator needs to run on host network
+	if !instance.Spec.ExternalStorage.Enable &&
+		!r.IsNoobaaStandalone &&
+		shouldUseHostNetworking(instance) {
+		err := updateOpHostNetworkInRookCephCSV(ctx, r.Client, instance.Namespace)
+		if err != nil {
+			r.Log.Error(err, "Failed to update Rook Ceph Operator HostNetwork.", "StorageCluster", klog.KRef(instance.Namespace, instance.Name))
+			return reconcile.Result{}, err
+		}
+	}
```

Comment:
Why is this here and does not follow the usual ensureCreated/ensureDeleted standard this repo standardised on?

Reply:
was not hitting test case errors because the code was trying to check a specific use case, 
but now with using ensureCreated, a lot of unit tests are failing because the code fails on finding rook-csv and deployment.

Do we want to just log a warning instead of throwing errors?